contracts/cheddar_5x5_tic_tac_toe/src/lib.rs

```diff
@@ -233,12 +246,42 @@ impl Contract {
         }
     }
 
-    pub fn get_last_move(&self, game_id: &GameId) -> (Option<Coords>, Piece){
+    pub fn get_last_move(&self, game_id: &GameId) -> (Option<Coords>, Piece, Option<GameResult>, Option<u32>){
+        let stored_game = self.stored_games.get(game_id);
+        if stored_game.is_some() {
+            let stored_game_unwraped = stored_game.unwrap();
+            if stored_game_unwraped.last_move.is_none() {
+                return (None, Piece::X, Some(stored_game_unwraped.game_result), None);
+            }
+            let (coords, piece) = stored_game_unwraped.last_move.unwrap();
+            return (Some(coords), piece, Some(stored_game_unwraped.game_result), None);
+        } else {
         let game = self.internal_get_game(game_id);
-        return (game.board.get_last_move(), game.current_piece.other());
+        let mut game_result = None;
+        if game.board.winner.is_some() {
+            let winner = game.board.winner.clone().unwrap();
+            game_result = match winner {
+                Winner::O => Some(GameResult::Win(game.players.0)),
+                Winner::X => Some(GameResult::Win(game.players.1)),
+                Winner::Tie => Some(GameResult::Tie),
+            }
+        }
+        return (game.board.get_last_move(), game.current_piece.other(), game_result, Some(nano_to_sec(game.last_turn_timestamp)));
+    }
+    }
+    fn get_game_result(&self, winner_piece: Option<Winner>, game_id: &GameId) -> Option<GameResult> {
+        if winner_piece.is_none() {
+            return None;
+        }
+        let game = self.get_game(game_id);
+        return match winner_piece.unwrap() {
+            Winner::O => Some(GameResult::Win(game.player1)),
+            Winner::X => Some(GameResult::Win(game.player2)),
+            Winner::Tie => Some(GameResult::Tie)
+        }
     }
 
-    pub fn make_move(&mut self, game_id: &GameId, coords: Coords) -> Option<Winner> {
+    pub fn make_move(&mut self, game_id: &GameId, coords: Coords) -> Option<GameResult> {
```

Comment:
instead of dealing with `Option<GameResult` multiple times, maybe we should add `Ongoing` variant to `GameResult`?

Reply:
but on the other hand UI has to do two checks: 
```
if( result != undefined) {
  if( result == Player1) ....
}

```